bigbluebutton-html5/imports/ui/components/video-provider/video-element/component.jsx

```diff
@@ -0,0 +1,91 @@
+import React, { Component } from 'react';
+import { styles } from '../styles';
+
+class VideoElement extends Component {
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    let cssClass;
+
+    if (this.props.shared || !this.props.localCamera) {
+      cssClass = styles.sharedWebcamVideoLocal;
+    } else {
+      cssClass = styles.sharedWebcamVideo;
+    }
+    return (
+      <div className={`${styles.videoContainer} ${cssClass}`} >
+        { this.props.localCamera ?
+          <video id="shareWebcam" muted autoPlay playsInline />
+          :
+          <video id={`video-elem-${this.props.videoId}`} autoPlay playsInline />
+        }
+        <div className={styles.videoText}>
+          <div className={styles.userName}>{this.props.name}</div>
+        </div>
+      </div>
+    );
+  }
+
+  componentDidMount() {
+    const { videoId, localCamera } = this.props;
+
+    let tag;
+    if (localCamera) {
+      tag = document.getElementById('shareWebcam');
+    } else {
+      tag = document.getElementById(`video-elem-${videoId}`);
+    }
+
+    if (localCamera && this.props.onShareWebcam === 'function') {
+      this.props.onShareWebcam();
+    }
+
+    if (typeof this.props.onMount === 'function') {
+      this.props.onMount(videoId, localCamera, this.getVideoConstraints(), tag);
+    }
+  }
+
+  componentWillUnmount() {
+    if (typeof this.props.onUnmount === 'function') {
+      this.props.onUnmount(this.props.videoId);
+    }
+  }
+
+  getVideoConstraints() {
```

Comment:
```js
getVideoConstraints() {
  let videoConstraints = {
    width: {
      min: 320,
      max: 640,
    },
    height: {
      min: 240,
      max: 480,
    },
  };

  if (!navigator.userAgent.match(/Version\/[\d\.]+.*Safari/)) {
    videoConstraints.frameRate = { min: 5, ideal: 10, };
  }

 return videoConstraints;
}
```

And those constraints probably should be config.

Reply:
Let's leave this for another PR.

Reply:
You can swap the current code with the one above anyway, the config could be done on another PR.